types/common_fuzz_test.go

```diff
@@ -0,0 +1,293 @@
+package types
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/trailofbits/go-fuzz-utils"
+)
+
+func FuzzReverse(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		require.Equal(t, data, reverse(reverse(data)))
+	})
+}
+
+func GetTypeProvider(data []byte) (*go_fuzz_utils.TypeProvider, error) {
+	tp, err := go_fuzz_utils.NewTypeProvider(data)
+	if err != nil {
+		return nil, err
+	}
+
+	// Prevent nil fields.
+	err = tp.SetParamsBiases(0, 0, 0, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	return tp, nil
+}
+
+func FuzzRoundTripUint64StringSlice(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		var value, decJSON Uint64StringSlice
+		tp, err := GetTypeProvider(data)
```

Comment:
most of the functionality in this test is shared by all other tests in this file. is there an easy way to reuse and just pass in the type?

Reply:
Yes, I definitely agree. I spent some time trying to figure out a good way of doing this with generics but couldn't figure it out. I'm pretty new to golang. I will see what I can do.

Reply:
Okay. I think I figured it out. What do you think now?

Reply:
nice 👍